src/mscorlib/src/System/Diagnostics/Debug.Unix.cs

```diff
@@ -22,16 +18,11 @@ private static void ShowAssertDialog(string stackTrace, string message, string d
             }
             else
             {
-                // TODO: #3708 Determine if/how to put up a dialog instead.
-                var exc = new DebugAssertException(message, detailMessage, stackTrace);
-                if (!s_shouldWriteToStdErr) 
-                {
-                    // We always want to print out Debug.Assert failures to stderr, even if
-                    // !s_shouldWriteToStdErr, so if it wouldn't have been printed in
-                    // WriteCore (only when s_shouldWriteToStdErr), print it here.
-                    WriteToStderr(exc.Message);
-                }
-                throw exc;
+                // In Core, we do not show a dialog.
+                // Fail in order to avoid anyone catching an exception and masking
+                // an assert failure.
+                var ex = new DebugAssertException(message, detailMessage, stackTrace);
```

Comment:
Generally speaking, yes that should make it obvious that it's a System.Diagnostics.Debug.Assert which ended up triggering the failfast (as opposed to just a different kind of failfast).  Without the exception, I'm not sure the message is surfaced in a meaningful way in watson (outside of actually debugging the dump itself).

The big difference is that a lot of frames are 'ignored' when processing stack frames for watson, so a callstack like: "user1 calls user2 calls user3 calls framework1 which asserts", you might get a bucket that looks like 'FailFast_User3" because we skipped 'framework1'.  Instead we'd look like 'DebugAsssertException_User3' which makes it a bit more clear it was an assert instead of raw failfast.

In any case, if anyone wants to see the real bucketing behavior, simply generate a full crash dump with the new code and a crash dump using only a string (no exception), then run !analyze -v' in windbg.  The failure buckets should be pretty different.

(Though to be clear, I'm fine with either behavior.  I do think the dummy DebugAssertException will add clarity though.)